src/services/database.ts

```diff
@@ -55,40 +58,75 @@ export const initDb = (client: MongoClient) => {
   db = client.db(dbName);
 };
 
-export const findAssetsByChecksums = async (checksums: string[]) => {
+export const findAssetsByChecksums = (checksums: string[]) => {
   return db.collection<AssetDocument>(ASSETS_COLLECTION).find({ _id: { $in: checksums } });
 };
 
-export const findPagesByBuildId = async (buildId: string | ObjectId) => {
+export const findPagesByBuildId = (buildId: string | ObjectId) => {
   const id = new ObjectId(buildId);
   const query = { build_id: id };
   return db.collection<PageDocument>(PAGES_COLLECTION).find(query);
 };
 
-export const findPagesByProject = async (project: string, branch: string) => {
+export const findPagesByProj = (project: string, timestamp?: number) => {
+  const pageIdQuery = getPageIdQuery(project);
+  const query: Filter<UpdatedPageDocument> = { page_id: pageIdQuery };
+  if (timestamp) {
+    const lastQuery = new Date(timestamp);
+    query['updated_at'] = { $gt: lastQuery };
+  }
+  console.log(query);
+  return db.collection<UpdatedPageDocument>(UPDATED_PAGES_COLLECTION).find(query);
+};
+
+export const findPagesByProjAndBranch = (project: string, branch: string) => {
   const pageIdQuery = getPageIdQuery(project, branch);
   const query = { page_id: pageIdQuery };
   return db.collection<UpdatedPageDocument>(UPDATED_PAGES_COLLECTION).find(query);
 };
 
-export const findUpdatedPagesByProject = async (project: string, branch: string, timestamp: number) => {
+export const findUpdatedPagesByProjAndBranch = (project: string, branch: string, timestamp: number) => {
   const pageIdQuery = getPageIdQuery(project, branch);
   const updatedAtQuery = new Date(timestamp);
   const query = { page_id: pageIdQuery, updated_at: { $gt: updatedAtQuery } };
   return db.collection<UpdatedPageDocument>(UPDATED_PAGES_COLLECTION).find(query);
 };
 
-export const findOneMetadataByBuildId = async (buildId: string | ObjectId) => {
+export const findMetadataByBuildId = (buildId: string | ObjectId) => {
   const id = new ObjectId(buildId);
   const query = { build_id: id };
-  return db.collection(METADATA_COLLECTION).findOne(query);
+  return db.collection(METADATA_COLLECTION).find(query);
 };
 
-export const findLatestMetadata = async (project: string, branch: string) => {
-  const filter = { project, branch };
-  const res = await db.collection(METADATA_COLLECTION).find(filter).sort('created_at', -1).limit(1).toArray();
-  if (!res || res.length !== 1) {
-    return null;
+/**
+ * Returns all metadata documents for a given project
+ * @param project
+ * @param lastQuery
+ * @returns
+ */
+export const findLatestMetadataByProj = (project: string, timestamp?: number) => {
+  const matchFilter: Filter<Document> = { project: project };
+  if (timestamp) {
+    const lastQuery = new Date(timestamp);
+    matchFilter['created_at'] = { $gt: lastQuery };
   }
-  return res[0];
+  const aggregationStages = [
```

Comment:
nice aggregations! 😎 

as a side note, it seems like a good idea to index the collections on `created_at`

Reply:
We definitely should! Thinking this could be applied directly or programmatically included in the persistence module's code